pkg/webhook/server/generic/pod/gracedelete/lifecycle_adapter.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2023 The KusionStack Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gracedelete
+
+import (
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"kusionstack.io/operating/pkg/controllers/utils/podopslifecycle"
+)
+
+var (
+	OpsLifecycleAdapter = &GraceDeleteOpsLifecycleAdapter{}
+)
+
+// PodDeleteOpsLifecycleAdapter tells PodOpsLifecycle the Pod deletion ops info
+type GraceDeleteOpsLifecycleAdapter struct {
+}
+
+// GetID indicates ID of one PodOpsLifecycle
+func (a *GraceDeleteOpsLifecycleAdapter) GetID() string {
+	return "grace-delete"
```

Comment:
Can we reuse the existing `delete` adapter?
Because the `grace-delete` here can share the same lifecycle progress with `delete`.

Reply:
The existing delete adapter is resued instead of `GraceDeleteOpsLifecycleAdapter`.

Reply:
The existing delete adapter is resued instead of `GraceDeleteOpsLifecycleAdapter`.